src/Tokenizer/AbstractTypeTransformer.php

```diff
@@ -21,76 +21,47 @@
  */
 abstract class AbstractTypeTransformer extends AbstractTransformer
 {
+    private const TYPE_END_TOKENS = [')', [T_CALLABLE], [T_NS_SEPARATOR], [T_STRING], [CT::T_ARRAY_TYPEHINT]];
+
+    private const TYPE_TOKENS = [
+        '|', '&', '(',
+        ...self::TYPE_END_TOKENS,
+        [CT::T_TYPE_ALTERNATION], [CT::T_TYPE_INTERSECTION], // some siblings may already be transformed
+        [T_WHITESPACE], [T_COMMENT], [T_DOC_COMMENT], // technically these can be inside of type tokens array
+    ];
```

Comment:
Dumb question, I just want to understand as I still don't know the codebase much: why some items are simple strings, and some types are wrapped as array? They weren't like that before your changes and I don't see any specific logic for it.

Reply:
This is related to how [`token_get_all`](https://www.php.net/manual/en/function.token-get-all.php#refsect1-function.token-get-all-returnvalues) returns tokens - sometimes it is a string, sometimes an array.